pkg/controller/actuator_delete.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"time"
+
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+
+	resourcemanager "github.com/gardener/gardener-resource-manager/pkg/manager"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+// Delete implements ContainerRuntime.Actuator.
+func (a *actuator) Delete(ctx context.Context, cr *extensionsv1alpha1.ContainerRuntime, cluster *extensionscontroller.Cluster) error {
```

Comment:
This `Delete` function will delete the `RuntimeClass` resource as well as the `DaemonSet` in the shoot. However, the installed gvisor binaries will remain on the node. But as the `RuntimeClass` was removed they aren't use anyways any longer, or should we remove them as well (if, how)?

Reply:
Can we have a small unit test for this function, please?

Reply:
What do you want me to check here?

I added an uninstall script, which will remove the binaries and delete the configurations of GVisor in the config.toml
when the pod is deleted, so it will clean the machine.

Reply:
You can test whether the expected DELETE calls happen for the managed resources

Reply:
Tests added